Alright, figured it out! I recreated the scenario by making Vector2i use `int64_t` instead of `int32_t` and this was the failing output:
```log
D:\Godot\Github\godot\drivers\gles3\storage\texture_storage.cpp(2320) : error C2220: the following warning is treated as an error
D:\Godot\Github\godot\drivers\gles3\storage\texture_storage.cpp(2320) : warning C4701: potentially uninitialized local variable 'scale' used
```
So [line 2320](https://github.com/godotengine/godot/blob/b1371806ad3907c009458ea939bd4b810f9deb21/drivers/gles3/storage/texture_storage.cpp#L2320) was the culprit, not strictly because a default case was passed, but because it was using the Vector2i `*` operator against an `int` when it was expecting an `int64_t`. So even though there's an implicit conversion between `int32_t` and `int64_t`, this only appears to take effect in a context where `int32_t` was initialized. Changing "scale" to `int64_t` fixed the issue, even when uninitalized

So while the proposed change in this PR is technically appropriate, it wasn't actually reaching that stage in practice. Instead, I'll shift direction to your initial proposal and assign an error to the default cases

--- a/drivers/gles3/storage/texture_storage.cpp
+++ b/drivers/gles3/storage/texture_storage.cpp
@@ -2314,7 +2314,9 @@ Rect2i TextureStorage::_render_target_get_sdf_rect(const RenderTarget *rt) const
 			scale = 200;
 		} break;
 		default: {
-		}
+			ERR_PRINT("Invalid viewport SDF oversize, defaulting to 100%.");
+			scale = 100;
+		} break;
 	}
 
 	margin = (rt->size * scale / 100) - rt->size;
@@ -2391,6 +2393,7 @@ void TextureStorage::_render_target_allocate_sdf(RenderTarget *rt) {
 			scale = 25;
 		} break;
 		default: {
+			ERR_PRINT("Invalid viewport SDF scale, defaulting to 100%.");
 			scale = 100;
 		} break;
 	}

--- a/servers/rendering/renderer_rd/storage_rd/texture_storage.cpp
+++ b/servers/rendering/renderer_rd/storage_rd/texture_storage.cpp
@@ -3511,7 +3511,9 @@ Rect2i TextureStorage::_render_target_get_sdf_rect(const RenderTarget *rt) const
 			scale = 200;
 		} break;
 		default: {
-		}
+			ERR_PRINT("Invalid viewport SDF oversize, defaulting to 100%.");
+			scale = 100;
+		} break;
 	}
 
 	margin = (rt->size * scale / 100) - rt->size;
@@ -3603,6 +3605,7 @@ void TextureStorage::_render_target_allocate_sdf(RenderTarget *rt) {
 			scale = 25;
 		} break;
 		default: {
+			ERR_PRINT("Invalid viewport SDF scale, defaulting to 100%.");
 			scale = 100;
 		} break;
 	}

